Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControl.cpp

```diff
@@ -90,32 +90,44 @@ namespace ROS2PoseControl
         }
     }
 
-    AZ::Outcome<AZ::Transform, const char*> ROS2PoseControl::GetCurrentTransformViaTF2()
+    AZ::Outcome<AZ::Transform, const char*> ROS2PoseControl::GetCurrentTransformViaTF2(
+        const AZStd::string& targetFrame,
+        const AZStd::string& sourceFrame)
     {
         geometry_msgs::msg::TransformStamped transformStamped;
         std::string errorString;
+        bool exceptionThrown = false;
+        const auto targetFrameCStr = targetFrame.c_str();
+        const auto sourceFrameCStr = sourceFrame.c_str();
         if (m_tf_buffer->canTransform(
-            m_configuration.m_referenceFrame.c_str(),
-            m_configuration.m_targetFrame.c_str(),
+            targetFrameCStr,
+            sourceFrameCStr,
             tf2::TimePointZero,
             &errorString))
         {
-            transformStamped = m_tf_buffer->lookupTransform(
-                m_configuration.m_referenceFrame.c_str(),
-                m_configuration.m_targetFrame.c_str(),
-                tf2::TimePointZero);
-            m_tf2WarningShown = false;
+            try
+            {
+                transformStamped = m_tf_buffer->lookupTransform(
+                    targetFrameCStr,
+                    sourceFrameCStr,
+                    tf2::TimePointZero);
+                m_tf2WarningShown = false;
+            } catch (const tf2::TransformException& ex)
+            {
+                errorString = ex.what();
+                exceptionThrown = true;
```

Comment:
Why doesn't you return here with ex.what() as fail?

Reply:
I wanted to have neat debug print